pubtools/pulplib/_impl/model/repository/base.py

```diff
@@ -597,18 +597,38 @@ def sync(self, options=None):
             )
         )
 
-    def lock(self, duration=LOCK_EXPIRATION_OFFSET, retry_wait=AWAIT_ACTIVATION_SLEEP):
+    def lock(self, context, duration=None):
         """
-        Lock the repo to show other tasks that it is being worked on.
-        This does not prevent modifications to the repo with the Pulp API, it
-        just provides a way to pause tasks to prevent race conditions.
-
+        Obtain an exclusive advisory lock on this repository.
+
+        Returns a context manager representing the lock, intended to be used
+        via a `with` statement. When the context is entered, the caller will
+        wait until the lock can be acquired (or raise an exception if the lock
+        can't be acquired).
+
+        Only a single :class:`~pubtools.pulplib.Client` is able to hold the lock
+        on a repository at any given time. The lock does not prevent modifications
+        to the repo with the Pulp API, and does not affect other Pulp client
+        implementations or instances of :class:`~pubtools.pulplib.Client` not
+        using the `lock` method.
         Args:
+            context:
+                A short description of the task being carried out with the lock.
+
+                This value will be added to the lock in the repo and may be
+                used for debugging.
+
             duration
-                How long the lock should remain valid for in seconds.
-            retry_wait:
-                How long to wait between attempting to activate the lock in
-                seconds.
+                 Maximum duration of the lock, in seconds.
```

Comment:
Seems like an off-by-one whitespace error here, this will probably mess with the docs rendering (should be possible to check by `tox -e docs`).

```suggestion
                Maximum duration of the lock, in seconds.
```